### PR TITLE
Pull github activity and create posts

### PR DESCRIPTION
**Pull request by kalegar**

### CODE REVIEW

File: distributedsocialnetwork/author/retrieval.py [hunk 1]
```diff
@@ -199,3 +200,126 @@ def get_detailed_author(author_id):
...
+                        visibility= 'PRIVATE'
+                    )
+                    post.origin = origin
+                    post.source = settings.FORMATTED_HOST_NAME + \
+                        'posts/' + str(post.id)
+                    post.save()
+                except:
+                    print("Failed to import github event!")
+                    
+def parse_github_event(event):
```
Comment (kalegar) [inline distributedsocialnetwork/author/retrieval.py:248]: This function handles parsing most common github activity events and returns suitable post content. We could cover more event types if needed. Any event types that aren't explicitly handled just return a simple string like "'user' in 'repo'".

File: distributedsocialnetwork/author/retrieval.py [hunk 2]
```diff
@@ -199,3 +200,126 @@ def get_detailed_author(author_id):
...
+
+# Get the github activity of the given author.
+# Creates private posts for every new github
+# event, as authors should only be able to
+# see their own github activity stream.
+def get_github_activity(author_id):
+    author = get_object_or_404(Author, id=author_id)
+    # If the author has set a github profile URL
+    if author.github:
+        githubUsername = author.github.rsplit('/', 1)[-1]
```
Comment (kalegar) [inline distributedsocialnetwork/author/retrieval.py:212]: This assumes the author's github profile url is valid, ie it ends with their github username

**Review by ColonelSanders21 — APPROVED** [review]
Looks good!

File: distributedsocialnetwork/distributedsocialnetwork/updates.py [hunk 3]
```diff
@@ -22,6 +22,10 @@ def get_all_visible_posts():
     author = Author.objects.filter(host=settings.FORMATTED_HOST_NAME)[0]
     get_visible_posts(author.id)
 
+def get_all_github_activity():
+    # Get the github activity for all our local authors.
+    for author in Author.objects.all.filter(origin__icontains=settings.FORMATTED_HOST_NAME):
```
Comment (ColonelSanders21) [inline distributedsocialnetwork/distributedsocialnetwork/updates.py]: Couple of typos here that fail to load the posts. Change to `Author.objects.filter` (remove the all) and switch `origin__icontains=settings.FORMATTED... to `host=settings.FORMATTED...`

**Review by ColonelSanders21 — CHANGES_REQUESTED** [review]
Couple of minor typos, when I fixed them it seems to work well

File: distributedsocialnetwork/distributedsocialnetwork/updates.py (hunk 3 above)
Comment (kalegar) [reply distributedsocialnetwork/distributedsocialnetwork/updates.py]: Thanks! Fixed.

**Review by ColonelSanders21 — APPROVED** [review]

**Pull request merged**